### PR TITLE
OKD Provider

### PR DESCRIPTION
**Pull request by skatlapa**

### CODE REVIEW

**Comment by skatlapa** [comment]
@arilivigni @robnester-rh Please review, thanks. I am not able to add reviewers, maybe due to perms

File: src/org/centos/contra/Infra/Utils.groovy [hunk 1]
```diff
@@ -299,6 +300,91 @@ def createOpenstackInstances(HashMap<String, String>openstackData){
     return openstackInstances
 }
 
+/**
+ * Method to create a collection of instances of the OpenShift Cloud Platform provider type based on supplied data
+ * @param openshiftData
+ * @return
+ */
+def createOpenshiftInstances(HashMap<String, String>openshiftData){
```
Comment (robnester-rh) [inline src/org/centos/contra/Infra/Utils.groovy]: We should talk about this method in more detail.

File: src/org/centos/contra/Infra/Utils.groovy [hunk 2]
```diff
@@ -497,6 +587,8 @@ static def getInstanceFromContext(instance, context) {
     return instance_context
 }
 
+//(skatlapa): Check if instance context is needed for OCP
```
Comment (robnester-rh) [inline src/org/centos/contra/Infra/Utils.groovy]: Remove comments

File: src/org/centos/contra/Infra/Utils.groovy [hunk 3]
```diff
@@ -299,6 +300,83 @@ def createOpenstackInstances(HashMap<String, String>openstackData){
...
+        instance.name = instance.name ?: openshiftData.name
+        instance.namespace = instance.namespace ?: openshiftData.namespace
+        instance.password = instance.password ?: openshiftData.password
+        instance.username = instance.username ?: openshiftData.username
+
+        // 1. set api* values here
+        // 2. check if value for Loadbalancer has to be set(neeed in provider class too)
+        // 3. check if keyPair is required for ocp
+        // 4. "" userData, user
+        def user = null
```
Comment (robnester-rh) [inline src/org/centos/contra/Infra/Utils.groovy]: Lines 322-325 can be rewritten as:

```` groovy
instance.user = instance.user ?: openshiftData.user
````

File: config/s2i/linchpin/linchpin-buildconfig-template.yaml [hunk 4]
```diff
@@ -27,7 +27,7 @@ objects:
     - type: ConfigChange
     source:
       type: DockerFile
-      dockerfile: "FROM contrainfra/linchpin-executor:v1.1.0"
+      dockerfile: "FROM contrainfra/linchpin-executor:v1.6.5rc1"
```
[the remark this answers is not in the document]
Comment (skatlapa) [reply config/s2i/linchpin/linchpin-buildconfig-template.yaml]: Ack, I spoke with Ari, we currently decide to follow name conventions as per LP release, so I'll update accordingly, currently, tag would mostly be renamed to: 1.6.5

File: resources/org/centos/contra/Infra/topologyTemplates/openshift.template [hunk 5]
```diff
@@ -0,0 +1,23 @@
+${index}_${providerType}:
```
Comment (skatlapa) [inline resources/org/centos/contra/Infra/topologyTemplates/openshift.template]: this file may have more iterations to follow, depending on how well the LP provider is tested, I'll update on my working branch and squash all the final commits when we have the image updated

File: src/org/centos/contra/Infra/Utils.groovy (hunk 3 above)
Comment (skatlapa) [reply src/org/centos/contra/Infra/Utils.groovy]: good to know, ack, groovy truth, thanks!

**Comment by robnester-rh** [comment]
@skatlapa is this PR complete?

**Comment by skatlapa** [comment]
This needs to be checked against the new LP image(which is all set). Sorry for delays, the folks who are blocked by this, could you test in your local env with the PR[1] and let me know of any bugs? Regardless, I will mostly address this by end of this week (Friday) 

[1] - https://github.com/skatlapa/contra-hdsl/tree/contra_583_gh

**Comment by skatlapa** [comment]
Update: I am currently testing this in my local env. Will give an ack when done(in a day or two).

**Comment by skatlapa** [comment]
Blocked by this Linchpin issue https://github.com/CentOS-PaaS-SIG/linchpin/issues/921

**Pull request closed without merging**